workspaces/src/network/account.rs

```diff
@@ -40,7 +102,191 @@ impl Contract {
         &self.account.id
     }
 
+    pub fn as_account(&self) -> &Account {
+        &self.account
+    }
+
     pub(crate) fn signer(&self) -> &InMemorySigner {
         self.account.signer()
     }
+
+    /// Call the current contract's function using the contract's own account
+    /// details to do the signing. Returns a Builder object that we will make
+    /// use to populate the rest of the call details.
+    ///
+    /// If we want to make use of the contract's account to call into a
+    /// different contract besides the current one, use
+    /// `contract.as_account().call` instead.
+    pub fn call<'a, T: Network>(
+        &self,
+        worker: &'a Worker<T>,
+        function: &str,
+    ) -> CallBuilder<'a, T> {
+        self.account.call(worker, self.id().clone(), function)
+    }
+
+    /// Call a view function into the current contract. Returns a result that
+    /// yields a JSON string object.
+    pub async fn view<T: Network>(
+        &self,
+        worker: &Worker<T>,
+        function: String,
```

Comment:
this is asymmetric with the signature above, we should probably do the same here.